sdk/eventhub/Azure.Messaging.EventHubs/src/EventData.cs

```diff
@@ -62,7 +76,14 @@ public class EventData
         ///   empty.
         /// </remarks>
         ///
-        public IReadOnlyDictionary<string, object> SystemProperties { get; }
+        public IReadOnlyDictionary<string, object> SystemProperties
+        {
+            get
+            {
+                Interlocked.CompareExchange(ref _systemProperties, new Dictionary<string, object>(), null);
```

Comment:
Should we have a static-empty-read-only dictionary to return in the default case?

Reply:
Interesting idea.   Avoid allocating on read for events that weren't read from the service.   I like it.